### PR TITLE
Assume all swap memory is free on values overflow

### LINKED ISSUE #6271: Unexpectedly huge byte values returned for system.memory.swap.used.bytes
6.1

Metricbeat is returning ginormous byte values for swap used.  There are docker and kube processes running on this bare metal server (metricbeat is running directly on the host machine, not within containers).

```
[2018-01-31T17:07:47,804][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch.
{"type"=>"mapper_parsing_exception", "reason"=>"failed to parse [system.memory.swap.used.bytes]", "caused_by"=>{"type"=>"i_o_exception", "reason"=>"Numeric value (18446744070278606848) out of range of long (-9223372036854775808 - 9223372036854775807)\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@45bfe490; line: 1, column: 342]"}}}}}
```

Workaround is to filter out fields using a processor with these large values.  If this is not something we can avoid because of golang, it will be nice for beats to filter these invalid values automatically without manual intervention.

CentOS Linux release 7.4.1708 (Core)

```
 free -m
              total        used        free      shared  buff/cache   available
Mem:         257675        9188      244749          27        3737      247391
Swap:          4083           0        4083
```

```
top - 10:38:10 up 3 days,  3:45,  4 users,  load average: 0.53, 0.70, 0.54
Tasks: 895 total,   1 running, 894 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.7 us,  0.9 sy,  0.0 ni, 98.4 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem : 26385961+total, 25061606+free,  9416044 used,  3827500 buff/cache
KiB Swap:  4182012 total,  4182012 free,        0 used. 25332158+avail Mem
```

### PR DESCRIPTION
**Pull request by jsoriano**
There have been reports about user swap memory being a value that looks
like an overflow. At the moment used swap memory is calculated as the
value of total memory minus the free memory, so this means that free
memory is bigger than total in some cases. This isn't possible in
principle. Assume that no swap is being used if that happens.

Closes #6271

### CODE REVIEW

File: libbeat/metric/system/memory/memory.go [hunk 1]
```diff
@@ -69,6 +69,14 @@ func GetSwap() (*SwapStat, error) {
 		return nil, err
 	}
 
+	// This shouldn't happen, but it has been reported to happen and
+	// this can provoke too big values for used swap.
+	// Workaround this by assuming that all swap is free in that case.
+	if swap.Free > swap.Total || swap.Used > swap.Total {
```
Comment (ruflin) [inline libbeat/metric/system/memory/memory.go]: I would suggest to at leats log something on the debug level in this cases. So if we want to investigate this further in the future, we can see the actual values reported.

Comment (sayden) [reply libbeat/metric/system/memory/memory.go]: Agree

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -38,6 +38,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha1...master[Check the HEAD d
 - Fix autodiscover configurations stopping when metadata is missing. {pull}8851[8851]
 - Log events at the debug level when dropped by encoding problems. {pull}9251[9251]
 - Refresh host metadata in add_host_metadata. {pull}9359[9359]
+- Handle cases of free swap being bigger than total swap by assuming no swap is being used. {issue}6271[6271] {pull}[]
```
Comment (sayden) [inline CHANGELOG.asciidoc]: You haven't written the pull number. Maybe I'd add the module involved too

**Comment by jsoriano** [comment]
Thanks for the reviews, changes addressed.

File: libbeat/metric/system/memory/memory.go [hunk 3]
```diff
@@ -69,6 +70,17 @@ func GetSwap() (*SwapStat, error) {
...
+		logp.Debug("memory",
+			"Odd values for swap memory, total: %v free: %v used: %v. Assuming all swap is free.",
```
Comment (ruflin) [inline libbeat/metric/system/memory/memory.go]:
```suggestion
			"Unexpected values for swap memory- total: %v free: %v used: %v.  Setting swap used to 0.",
```

**Pull request merged**